src/main/java/com/demo/classroom/Controller/DashboardController.java

```diff
@@ -0,0 +1,69 @@
+package com.demo.classroom.Controller;
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.demo.classroom.DTO.ApiResponse;
+import com.demo.classroom.DTO.CourseDTO;
+import com.demo.classroom.Service.CourseService;
+
+import jakarta.validation.Valid;
+
+import java.util.HashMap; // Make sure to import this
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.BindingResult;
+
+@RestController
+public class DashboardController {
+
+    @Autowired
+    private CourseService courseService;
```

Comment:
The @Autowired annotation is used here to inject dependencies, but constructor injection is generally considered a better practice.

Reply:
done